emit_translations.py

```diff
@@ -44,13 +44,25 @@ def translate_src_files(src_dir: str, out_dir: str, translations: dict[Macro, st
         startLine = int(startDefLocParts[1]) - 1
         endLine = int(endDefLocParts[1]) - 1
 
+        endLineContent = src_file_content[endLine]
+
+        # Some code bases may define macros with an opening comment on the last line,
+        # preserve it here
+        if '/*' in endLineContent.strip() and '*/' not in endLineContent.strip():
+                endLineComment = '/*' + endLineContent.split('/*', 1)[1]
+        else:
+            endLineComment = ''
```

Comment:
I would initialize `endLineComment` before the if branch and outside of any nested scope, e.g.,
```suggestion
        endLineComment = ''
        if '/*' in endLineContent.strip() and '*/' not in endLineContent.strip():
                endLineComment = '/*' + endLineContent.split('/*', 1)[1]
```